ionoscloud/resource_server_boot_device_selection.go

```diff
@@ -0,0 +1,153 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/internal/uuidgen"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi/cloudapiserver"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceServerBootDeviceSelection() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServerBootDeviceSelectionCreate,
+		ReadContext:   resourceServerBootDeviceSelectionRead,
+		UpdateContext: resourceServerBootDeviceSelectionUpdate,
+		DeleteContext: resourceServerBootDeviceSelectionDelete,
+		CustomizeDiff: checkBootDeviceSelectorImmutableFields,
+		Schema: map[string]*schema.Schema{
+
+			"datacenter_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the Datacenter that holds the server for which the boot volume is selected",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"server_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the Server for which the boot device will be selected.",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"boot_device_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the entity to set as primary boot device. Possible boot devices are CDROM Images and Volumes",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"default_boot_volume_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the first attached volume of the Server, which will be the default boot volume unless another is explicitly specified.",
+				Computed:    true,
+			},
+		},
+
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func checkBootDeviceSelectorImmutableFields(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
```

Comment:
Would probably be better off if we set forceNew on the fields that are immutable, like datacenter_id and server_id